src/tensorlake/function_executor/proto/function_executor.proto

```diff
@@ -161,6 +173,128 @@ message InfoResponse {
     optional string sdk_language_version = 4;
 }
 
+// === run_task_allocations_session RPC === //
+
+message SerializedObjectID {
+    optional string id = 1;
+}
+
+message SerializedObjectManifest {
+    optional SerializedObjectID id = 1;
+    optional SerializedObjectEncoding encoding = 2;
+    optional uint64 encoding_version = 3;
+    optional uint64 size = 4;
+    optional string sha256_hash = 5;
+    optional uint32 chunk_count = 6;
+}
+
+message SerializedObjectChunk {
+    // The ID of the serialized object this chunk belongs to.
+    optional SerializedObjectID id = 1;
+    // The chunk data.
+    optional bytes data = 2;
+    // The index of the chunk in the serialized object. 0 based.
+    optional uint32 index = 3;
+}
+
+// Used for uploading a serialized object from initiating side to the accepting side.
```

Comment:
It seems a little odd to have two completely different forms of serialized objects; can we combine them?